example/index.tsx

```diff
@@ -3,29 +3,32 @@ import * as React from 'react';
 import * as ReactDOM from 'react-dom';
 import MetaDiagram, {MetaNode, Position, ComponentsMap, MetaLink} from "./..";
 import {CustomNodeWidget} from "./components/widgets/CustomNodeWidget";
-import {makeStyles} from "@material-ui/core";
+import { makeStyles } from "@mui/styles";
 import CustomLinkWidget from "./components/widgets/CustomLinkWidget";
 import BG from "./components/assets/svg/bg-dotted.svg";
+import { ThemeProvider } from '@mui/material';
+import CssBaseline from '@mui/material/CssBaseline';
+import theme from './theme';
 
 const useStyles = makeStyles(_ => ({
-    main: {
-        position: 'absolute',
-        height: '100%',
-        width: '100%',
-    },
-    canvasBG: {
-        backgroundImage: `url(${BG})`
-    }
-}))
+  main: {
+    position: 'absolute',
+    height: '100%',
+    width: '100%',
+  },
+  canvasBG: {
+    backgroundImage: `url(${BG})`
+  }
+}));
 
 const App = () => {
     const classes = useStyles();
 
-    const node1 = new MetaNode('1', 'node1', 'default', new Position(250, 100),
-        new Map(Object.entries({color: 'rgb(0,192,255)'})))
+    const node1 = new MetaNode('1', 'Node 1', 'default', new Position(250, 100),
+        new Map(Object.entries({'variant': 'node-red'})))
 
-    const node2 = new MetaNode('2', 'node2', 'default', new Position(500, 100),
-        new Map(Object.entries({color: 'rgb(255,192,0)'})))
+    const node2 = new MetaNode('2', 'Node 2', 'default', new Position(500, 100),
```

Comment:
@afonsobspinto @vidhya-metacell I suggest to add variant to the interface as a parameter (like position) instead of using options. This gives a better usability and clearer programmatic interface. I suggest that we do that for all parameters that we "currently" know.

Reply:
@afonsobspinto could you please add this to the node model.